Move ENABLE_SWIFTUI flag to PlatformEnableCocoa.h

emw-apple opened the pull request:
#### 692497e37540b9a624c194afac3c2b76ca90cd8c
<pre>
Move ENABLE_SWIFTUI flag to PlatformEnableCocoa.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=297028">https://bugs.webkit.org/show_bug.cgi?id=297028</a>
<a href="https://rdar.apple.com/157706818">rdar://157706818</a>

Reviewed by Richard Robinson.

It was previously only defined in xcconfigs, which is an anti-pattern.
It&apos;s still useful to have an xcconfig definition in WebKit, because a
handful of build settings are keyed off of $(WK_ENABLE_SWIFTUI), but the
TestWebKitAPI usage can read directly from WTF.

With the definition moved, change -[WKWebView
_updateScrollGeometryWithContentOffset:contentSize:] to be a no-op in
builds where the SwiftUI API does not exist. This avoids confusion in
SPI checking by calling respondsToSelector for an internal delegate
methods that does not exist.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:

(-[WKWebView _updateScrollGeometryWithContentOffset:contentSize:]):
* Tools/TestWebKitAPI/Configurations/TestWebKitAPIBundle.xcconfig:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKScrollGeometryTests.mm: Only
run these tests when ENABLE_SWIFTUI is active.

Canonical link: <a href="https://commits.webkit.org/299034@main">https://commits.webkit.org/299034@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4870c2caa6718ea673b9fde80f102688381a1ff6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116932 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36597 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27171 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123017 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/68948 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4865d788-0f1e-46d4-8241-38de712d90dd) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118821 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37294 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45197 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88810 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43532 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d2a0f929-b8c5-4584-872a-accd8a6af02a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119881 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29745 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104902 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69269 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/eb9bebdf-2ac1-4e5f-a891-a65d46633b29) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28812 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23009 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66673 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/109048 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99133 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23163 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126144 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115460 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43832 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32943 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97472 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44197 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101104 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97275 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24862 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42606 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20546 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/40227 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43717 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49313 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144159 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43184 "Built successfully") | | [💥 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37124 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46523 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44889 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -818,6 +818,12 @@
 #endif
 
+#if !defined(ENABLE_SWIFTUI) \
+    && ((PLATFORM(MAC) && __MACOSX_OS_VERSION_MIN_REQUIRED >= 260000) \
```
should this also have visionOS in the list?

emw-apple replied:
Yeah, good catch!

emw-apple commented on Source/WebKit/Configurations/Base.xcconfig:
```diff
@@ -133,8 +133,9 @@ WK_ENABLE_SWIFTUI[sdk=appletv*] = NO;
-SWIFT_ACTIVE_COMPILATION_CONDITIONS_SWIFTUI_YES = ENABLE_SWIFTUI
+WK_SWIFTUI_COMPILATION_CONDITIONS_SWIFTUI_YES = ENABLE_SWIFTUI;
+GCC_PREPROCESSOR_DEFINITIONS = $(inherited) $(WK_SWIFTUI_COMPILATION_CONDITIONS_$(WK_ENABLE_SWIFTUI));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS = $(inherited) $(WK_SWIFTUI_COMPILATION_CONDITIONS_$(WK_ENABLE_SWIFTUI));
```
There is a typo here; we're expanding a variable that should be "YES" but checking for "SWIFTUI_YES". But it doesn't affect the build because the definition in PlatformEnableCocoa.h still works. Maybe I should just remove this snippet entirely…

emw-apple committed:
Move ENABLE_SWIFTUI flag to PlatformEnableCocoa.h
https://bugs.webkit.org/show_bug.cgi?id=297028
rdar://157706818

Reviewed by Richard Robinson.

It was previously only defined in xcconfigs, which is an anti-pattern.
It's still useful to have an xcconfig definition in WebKit, because a
handful of build settings are keyed off of $(WK_ENABLE_SWIFTUI), but the
TestWebKitAPI usage can read directly from WTF.

With the definition moved, change -[WKWebView
_updateScrollGeometryWithContentOffset:contentSize:] to be a no-op in
builds where the SwiftUI API does not exist. This avoids confusion in
SPI checking by calling respondsToSelector for an internal delegate
methods that does not exist.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:

(-[WKWebView _updateScrollGeometryWithContentOffset:contentSize:]):
* Tools/TestWebKitAPI/Configurations/TestWebKitAPIBundle.xcconfig:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKScrollGeometryTests.mm: Only
run these tests when ENABLE_SWIFTUI is active.

Canonical link: https://commits.webkit.org/299034@main
